zig-mode.el

```diff
@@ -123,6 +135,31 @@ If given a SOURCE, execute the CMD on it."
   (interactive)
   (zig--run-cmd "run" (buffer-file-name) "-O" zig-run-optimization-mode))
 
+(defun zig--find-project-root ()
```

Comment:
Really what I'd suggest instead of these changes is to add a (likely trivial) snippet that allows `project.el` to find the project root of Zig project. Then you can use `project-*` functions as needed in `zig-project-build-run` below.